test/plugin/example/plugin.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"net"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// NewConnector is a required method on the tcp.Plugin interface. It returns a
+// tcp.Connector.
+//
+// The single argument passed in is of type connector.Resources. It contains
+// connector-specific config and a logger.
+func NewConnector(conRes connector.Resources) tcp.Connector {
+	connectorFunc := func(
+		clientConn net.Conn,
+		credentialValuesByID connector.CredentialValuesByID,
+	) (backendConn net.Conn, err error) {
+		// singleUseConnector is responsible for generating the authenticated connection
+		// to the target service for each incoming client connection
+		singleUseConnector := &SingleUseConnector{
+			logger: conRes.Logger(),
+		}
+
+		return singleUseConnector.Connect(clientConn, credentialValuesByID)
+	}
+
+	return tcp.ConnectorFunc(connectorFunc)
+}
+
+// PluginInfo is required as part of the Secretless plugin spec. It provides
+// important metadata about the plugin.
+func PluginInfo() map[string]string {
+	return map[string]string{
+		"pluginAPIVersion": "0.1.0",
+		"type":             "connector.tcp",
+		"id":               "example-tcp-connector",
+		"description":      "it's just an example",
```

Comment:
Maybe something better/more formal should be used here (eg. `Example TCP Plugin`)